juicer/workflow/workflow.py

```diff
@@ -1,27 +1,263 @@
 import networkx as nx
-
+import matplotlib.pyplot as plt
+from juicer.service import tahiti_service
 
 class Workflow:
+    """
+        - Set and get Create a graph
+        - Identify tasks and flows
+        - Set and get workflow
+        - Add edges between tasks (source_id and targed_id)
+
+    """
+    WORKFLOW_DATA_PARAM = 'workflow_data'
+    WORKFLOW_GRAPH_PARAM = 'workflow_graph'
+    WORKFLOW_GRAPH_SORTED_PARAM = 'workflow_graph_sorted'
+    WORKFLOW_PARAM = 'workflow'
+    GRAPH_PARAM = 'graph'
+
+    WORKFLOW_GRAPH_SOURCE_ID_PARAM = 'source_id'
+    WORKFLOW_GRAPH_TARGET_ID_PARAM = 'target_id'
+
     def __init__(self, workflow_data):
-        self.graph = nx.DiGraph()
-        self.workflow = workflow_data
-        for task in workflow_data['tasks']:
-            self.graph.add_node(task.get('id'), attr_dict=task)
 
-        for flow in workflow_data['flows']:
-            self.graph.add_edge(flow['source_id'], flow['target_id'],
+        # Initialize
+        self.workflow_graph = nx.MultiDiGraph()
+
+        # Workflow dictionary
+        self.workflow_data = workflow_data
+
+        # Construct graph
+        self.workflow_graph = self.builds_initial_workflow_graph()
+
+        # self.number_of_outc = self.check_outdegree_edges()
+
+        # Topological sorted tasks according to their dependencies
+        self.sorted_tasks = []
+
+        # Verify null edges to topological_sorted_tasks
+        if self.check_null_target_id_tasks() \
+                and \
+            self.check_null_source_id_tasks():
+
+            self.sorted_tasks = self.get_topological_sorted_tasks()
+        else:
+            raise AttributeError(
+                "Parameter '{}/{}' must be informed for task {}".format(
+                    self.WORKFLOW_GRAPH_SOURCE_ID_PARAM,
+                    self.WORKFLOW_GRAPH_TARGET_ID_PARAM,
+                    self.__class__))
+
+
+
+    def builds_initial_workflow_graph(self):
+        """ Builds a graph with the tasks """
+
+        for task in self.workflow_data['tasks']:
+
+            query_tahiti = self.get_ports_from_operation_tasks(
+                                    task.get('operation')['id'])
+
+            self.workflow_graph.add_node(
+                task.get('id'),
+                tahiti_indegree_required=query_tahiti['N_INPUT'],
+                tahiti_indegree_multiplicity=query_tahiti['M_INPUT'],
+                tahiti_outdegree_required=query_tahiti['N_OUTPUT'],
+                tahiti_outdegree_multiplicity=query_tahiti['M_OUTPUT'],
+                attr_dict=task)
+
```

Comment:
Not all ports are required, Maybe we should add this information in the metadata.